src/client/clouds.cpp

```diff
@@ -170,8 +170,9 @@ void Clouds::render()
 
 	// Read noise
 
-	bool *grid = new bool[m_cloud_radius_i * 2 * m_cloud_radius_i * 2];
-
+	std::vector<char> grid(m_cloud_radius_i * 2 * m_cloud_radius_i * 2); // vector<bool> is broken
```

Comment:
It's likely not "broken" but rather [std::vector\<bool\>](https://en.cppreference.com/w/cpp/container/vector_bool) may use efficient bit-packing to store elements, which is not what you want here

Reply:
it is broken at least in the sense it is not a container

Reply:
My suggestion here is just to change the comment to say "vector<bool> doesn't work here" or similar.